tests/utils/node_launcher.go

```diff
@@ -0,0 +1,75 @@
+package utils
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/kurtosis-tech/kurtosis/api/golang/engine/lib/kurtosis_context"
+)
+
+const (
+	isPartitioningEnabled    = false
+	enclaveIdPrefix          = "test"
+	avalancheStarlarkPackage = "github.com/kurtosis-tech/avalanche-package"
+	// forces the node to launch on 9650 instead of ephemeral ports
+	forceExposeOn9650  = `{"test_mode": true}`
+	defaultParallelism = 4
+	firstNodeId        = "node-0"
+)
+
+func SpinupAvalancheNode() (string, func(), error) {
```

Comment:
What would it take to change this from `SpinupAvalancheNode` to spin up a network of N nodes and how would you specify configs of each individual node?

Reply:
To run multiple nodes; we'll have to change the args to include `node_count` like

```
forceExposeOn9650        = `{"test_mode": true, "node_count": 5}`
```

This would run 5 different nodes. Note Kurtosis by default tries to expose ephemeral ports but we're using the `"test_mode": true` to force public ports to be a certain number(first node rpc gets 9650, second gets 9652 and so forth). Ideally we remove the `test_mode` and pass around the ephemeral port generated inside these tests.

Re config - At the moment all nodes start with the same config except for bootstrap information. The second bootstraps from the first one; the third from the first two. What other config would you like to tweak? It shouldn't be too hard to set it; but I wanted to get something minimal out

Reply:
Cool that makes sense. AvalancheGo requires uses two ports the HTTP port (https://docs.avax.network/nodes/maintain/avalanchego-config-flags#--http-port-int) and the Staking port (https://docs.avax.network/nodes/maintain/avalanchego-config-flags#--staking-port-int).

If it's easier for Kurtosis to use ephemeral ports, you may just want to assign those two ports via these flags.

As in the documentation, the staking port is where the node listens for incoming p2p connections and the http port is used to run an API server.

Reply:
So we have one container per Avalanche Node inside the package. Within the Docker network the ports are set to the defaults 9650 and 9651; Kurtosis then produces (by default) an ephemeral port on localhost. That is the port that the tests interact with; with `test_mode` true I am forcing Kurtosis to produce local port mappings that are expected

So 9650,9651 on the first node will map to 9650 and 9651 on localhost
For the second avalanche node they map to 9652 and 9653
.. so on and so forth!

I am using `test_mode` as the tests seem to hardcode 

```
var (
	DefaultLocalNodeURI = "http://127.0.0.1:9650"
	NodeURIs            = []string{DefaultLocalNodeURI, "http://127.0.0.1:9652", "http://127.0.0.1:9654", "http://127.0.0.1:9656", "http://127.0.0.1:9658"}
)
```

and I didn't want to change your code base too much! There's not much extra effort for Kurtosis here!

Perhaps I can make the load test a multi node load test based on what I see here

```
tests/load/load_test.go
56:		rpcEndpoints := make([]string, 0, len(utils.NodeURIs))
57:		for _, uri := range []string{utils.DefaultLocalNodeURI} { // TODO: use NodeURIs instead, hack until fixing multi node in a network behavior
```

Reply:
I have changed the load test slightly and now I pass utils.NodeURIs

Reply:
I have reverted the multi node load test as it flaked on one run - https://github.com/h4ck3rk3y/subnet-evm/actions/runs/5093958012/jobs/9157181604